### ISSUE TITLE
v1.1 e2e failure against v1.2 master: Services should be able to change the type and nodeport settings of a service

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#21693]
This test is failing when running v1.1 of the test against a v1.2 master, (regardless of the version of the nodes):

**Services should be able to change the type and nodeport settings of a service**

*/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/service.go:585 Feb 22 06:28:54.826: got unexpected len(Status.LoadBalancer.Ingress) for NodePort service: &{{ } {mutability-service-test e2e-tests-services-yny7g /api/v1/namespaces/e2e-tests-services-yny7g/services/mutability-service-test 6a34e1cb-d970-11e5-a4db-42010af00002 3261 0 2016-02-22 06:27:34 -0800 PST map[] map[]} {ClusterIP [{ TCP 80 {0 80 } 0}] map[testid:mutability-service-test-6a31fb26-d970-11e5-8a01-42010af01555] 10.183.241.231 [] None} {{[{130.211.134.225 }]}}}*

Investigating...

**Comment by ihmccreery** [comment]
This just started failing Feb 18, 2016 10:22:29 PM.

**Comment by ihmccreery** [comment]
See http://kubekins.dls.corp.google.com/job/kubernetes-upgrade-gke-1.1-master-step4-e2e-old/ and http://kubekins.dls.corp.google.com/job/kubernetes-upgrade-gke-1.1-master-step6-e2e-old/

**Comment by bprashanth** [comment]
```
02:47:10   Feb 21 02:47:09.721: got unexpected len(Status.LoadBalancer.Ingress) for NodePort service: &{{ } {mutability-service-test  e2e-tests-services-h8905 /api/v1/namespaces/e2e-tests-services-h8905/services/mutability-service-test 513afa5c-d888-11e5-a8aa-42010af00005 4282 0 2016-02-21 02:46:09 -0800 PST <nil> <nil> map[] map[]} {ClusterIP [{ TCP 80 {0 80 } 0}] map[testid:mutability-service-test-513a8097-d888-11e5-8b98-42010af01555] 10.183.250.31 []  None} {{[{104.197.181.68 }]}}}
```

I think this is because we introduced a /status endpoint and the old test was just doing something wrong (it was checking the status.loadbalancer was 0 length right after the Service update, and the only reason that ever worked was because the Service update would clobber status -- which we started preventing with the /status endpoint). 

@caesarxuchao can you cherry-pick your test fix?

**Comment by caesarxuchao** [comment]
@bprashanth cherry-picking the test fix (https://github.com/kubernetes/kubernetes/commit/90b4662d8d02d98c2df4ed36e9f0b1e1e016517a#diff-20a4e2095b63ecd60dd25e78bcd67372R608) will break the test when running against 1.1 master. How about removing this check in 1.1?

**Comment by bprashanth** [comment]
That test should confirm that the lb is destroyed, and the service.ingress.status is cleared. As long as that's preserved I'm fine removing the check.

**Comment by caesarxuchao** [comment]
> That test should confirm that the lb is destroyed, and the service.ingress.status is cleared. 

These are verified after the `waitForLoadBalancerDestroy`. I'll just remove the check.

**Comment by caesarxuchao** [comment]
fixed by #21780. Closing.
